activation/nipost_test.go

```diff
@@ -281,73 +310,194 @@ func TestNIPostBuilder_BuildNIPost(t *testing.T) {
+	poets := make([]PoetProvingServiceClient, 0, 2)
+	poets = append(poets, createMockPoetService(t, []byte("poet0")))
+	poets = append(poets, createMockPoetService(t, []byte("poet1")))
+	nb := NewNIPostBuilder(minerID, &postSetupProviderMock{}, poets, poetDb, sql.InMemory(), logtest.New(t))
+
+	g, ctx := errgroup.WithContext(context.TODO())
+	resultChan := make(chan struct {
+		*types.NIPost
+		error
+	})
+	deadlineChan := make(chan struct{}, 1)
+
+	challenge := types.BytesToHash([]byte("challenge"))
+	g.Go(func() error {
+		nipost, _, err := nb.BuildNIPost(ctx, &challenge, deadlineChan)
+		resultChan <- struct {
+			*types.NIPost
+			error
+		}{nipost, err}
+		return nil
+	})
+
+	// Act
+	// Store proofMsg only from poet1
+	proofMsg := types.PoetProofMessage{
+		PoetServiceID: []byte("poet1"),
+		PoetProof: types.PoetProof{
+			Members:   [][]byte{challenge[:]},
+			LeafCount: 1,
+		},
+	}
+	ref, err := proofMsg.Ref()
+	assert.NoError(err)
+	poetDb.StoreProof(ref, &proofMsg)
+	time.Sleep(time.Millisecond * 10)
+
+	// Signal that the time is out
+	close(deadlineChan)
+
+	// Verify
+	result := <-resultChan
+	assert.NoError(result.error)
+	assert.Equal(*result.NIPost.Challenge, challenge)
+	proof, err := poetDb.GetProof(result.NIPost.PostMetadata.Challenge)
+	assert.NoError(err)
+	assert.EqualValues(proof.LeafCount, 1)
+}
+
+func TestNIPostBuilder_ManyPoETs_AllFinished(t *testing.T) {
+	t.Parallel()
+	// Arrange
+	assert := require.New(t)
+	poetDb := NewPoetDb(sql.InMemory(), logtest.New(t))
+	poets := make([]PoetProvingServiceClient, 0, 2)
+	poets = append(poets, createMockPoetService(t, []byte("poet0")))
+	poets = append(poets, createMockPoetService(t, []byte("poet1")))
+	nb := NewNIPostBuilder(minerID, &postSetupProviderMock{}, poets, poetDb, sql.InMemory(), logtest.New(t))
+
+	challenge := types.BytesToHash([]byte("challenge0"))
+	g, ctx := errgroup.WithContext(context.TODO())
+	resultChan := make(chan struct {
+		*types.NIPost
+		error
+	})
+	deadlineChan := make(chan struct{}, 1)
+
+	g.Go(func() error {
+		nipost, _, err := nb.BuildNIPost(ctx, &challenge, deadlineChan)
+		resultChan <- struct {
+			*types.NIPost
+			error
+		}{nipost, err}
+		return nil
+	})
+
+	// Act
+	// Store proofs from both poets
+	proofMsg := types.PoetProofMessage{
+		PoetServiceID: []byte("poet0"),
+		PoetProof: types.PoetProof{
+			Members:   [][]byte{challenge[:]},
+			LeafCount: 4,
+		},
+	}
+	ref, err := proofMsg.Ref()
+	assert.NoError(err)
+	poetDb.StoreProof(ref, &proofMsg)
+
+	proofMsg = types.PoetProofMessage{
+		PoetServiceID: []byte("poet1"),
+		PoetProof: types.PoetProof{
+			Members:   [][]byte{challenge[:]},
+			LeafCount: 1,
+		},
+	}
+	ref, err = proofMsg.Ref()
+	assert.NoError(err)
+	poetDb.StoreProof(ref, &proofMsg)
+
+	// Verify
+	result := <-resultChan
+	assert.NoError(result.error)
+	assert.Equal(*result.NIPost.Challenge, challenge)
+	proof, err := poetDb.GetProof(result.NIPost.PostMetadata.Challenge)
```

Comment:
This can be simplified to:

```suggestion
	result := <-resultChan
	assert.Equal(challenge, *result.Challenge)
	proof, err := poetDb.GetProof(result.PostMetadata.Challenge)
```